pkg/hsskey/filereader.go

```diff
@@ -129,24 +131,24 @@ func readHssFromFile(filePath string) ([][]byte, error) {
 	return hssList, nil
 }
 
-func getHssFromFile(verbose bool, verboseDangerous bool, filePaths []string) ([][]byte, error) {
+// GetHssFromFile reads HSS keys from the specified files.
+// Each HSS key is 64 bytes long and has a checksum for validation. Duplicate HSS are removed.
+func GetHssFromFile(warnings io.Writer, verboseDangerous bool, filePaths []string, minHssPerFile int) ([][]byte, error) {
 	allHss := [][]byte{}
 	for _, f := range filePaths {
-		hssKeys, err := readHssFromFile(f)
+		hssKeys, err := ReadHssFromFile(f, minHssPerFile)
 		if err != nil {
 			log.Printf("Failed to read HSS keys from file %s. err: %v", f, err)
 			continue
 		}
 
-		if verbose {
+		if verboseDangerous && warnings != nil {
 			msg := fmt.Sprintf("Reading HSS keys from file=%s", f)
-			if verboseDangerous {
-				for _, hss := range hssKeys {
-					msg = msg + fmt.Sprintf("\nseed=%x, seed(octal escape sequence)=%s", hss,
-						toOctalEscapeSequence(hss))
-				}
+			for _, hss := range hssKeys {
+				msg = msg + fmt.Sprintf("\nseed=%x, seed(octal escape sequence)=%s", hss,
+					toOctalEscapeSequence(hss))
 			}
-			log.Print(msg)
+			io.WriteString(warnings, msg)
```

Comment:
io.WriteString does not append a new-line, but log.Print does. That's why the test is failing now

Reply:
Duh! Thanks Chris! I had tested io.Write but for some reason I skipped testing WriteString mistaking taking the \n for granted. Fixed.